File: android/src/main/java/org/lichess/plugin/SoundEffectPlugin.kt
package org.lichess.plugin

import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin

@CapacitorPlugin(name = "SoundEffect")
class SoundEffectPlugin : Plugin() {

  private val implementation = new SoundEffect()

  @PluginMethod
  fun loadSound(call: PluginCall) {
    val audioId = call.getString("id")
    val path = call.getString("path")

    if (audioId === null) {
      call.error("Must supply an id")
      return
    }
    if (path === null) {
      call.error("Must supply a path")
      return
    }

    implementation.loadSound(audioId, path)
    call.resolve()
  }

  @PluginMethod
  fun play(call: PluginCall) {
    val audioId = call.getString("id")

    if (audioId === null) {
      call.error("Must supply an id")
      return
    }

    implementation.play(audioId)
    call.resolve()
  }
}

File: ios/Plugin/SoundEffectPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(SoundEffectPlugin)
public class SoundEffectPlugin: CAPPlugin {
    

    private let implementation = SoundEffect()

    @objc func loadSound(_ call: CAPPluginCall) {
        guard let path = call.options["path"] as? String else {
            call.reject("Must provide a path")
            return
        }
        
        guard let audioId = call.options["id"] as? String else {
            call.reject("Must provide an id")
            return
        }

        do {
            try implementation.loadSound(audioId: audioId, path: path)
        } catch {
            call.reject("Could not load file")
            return
        }
    }
    
    @objc func play(_ call: CAPPluginCall) {
        guard let audioId = call.options["id"] as? String else {
            call.reject("Must provide an id")
            return
        }
            
        do {
            try implementation.play(audioId)
        } catch {
            call.reject("Audio not found")
        }
        
        call.resolve()
    }

}
